File: C#/19. Remove Nth Node From End of List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution
{
    public ListNode RemoveNthFromEnd(ListNode head, int n)
    {
        int count = 0;
        ListNode temp = head;
        while (temp != null)
        {
            temp = temp.next;
            ++count;
        }
        if (count == n)
            return head.next;
        temp = head;
        for (int i = 1; i < count - n; ++i)
            temp = temp.next;
        temp.next = temp.next.next;
        return head;
    }
}

File: python/19. Remove Nth Node From End of List.py
# -*- coding: utf-8 -*-

"""
@Author: xiezizhe
@Date: 24/3/2020 下午11:32
"""


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        first, second = head, head
        while n > 0 and second is not None:
            second = second.next
            n -= 1
        if second is None:
            return head.next

        while second.next is not None:
            second = second.next
            first = first.next

        first.next = first.next.next
        return head
